runtime.go

```diff
@@ -1424,8 +1424,7 @@ func (r *Runtime) CaptureCallStack(depth int, stack []StackFrame) []StackFrame {
 
 // Interrupt a running JavaScript. The corresponding Go call will return an *InterruptedError containing v.
 // Note, it only works while in JavaScript code, it does not interrupt native Go functions (which includes all built-ins).
-// If the runtime is currently not running, it will be immediately interrupted on the next Run*() call.
-// To avoid that use ClearInterrupt()
+// The runtime will be immediately interrupted on any future Run*() call until ClearInterrupt() is called.
```

Comment:
Have you considered adding ClearInterrupt() to (*Runtime).leave()? I think this would solve the issue without changing the specified behaviour.

Reply:
I haven't, and I can't figure out how it will be broken, so I did it and just tested it - it seems to work as expected even with our original problem :tada: 
Unfortunately it also seem like it doesn't get called as it seems like leave isn't called after the recover [here](https://github.com/dop251/goja/blob/189bfeb9f530375f599387cac5bb9109c99dc813/runtime.go#L1367) or [here](https://github.com/dop251/goja/blob/189bfeb9f530375f599387cac5bb9109c99dc813/runtime.go#L2329) :thinking: 

So it will likely need even more changes :( 

I will try to look into it more tomorrow, thanks for the feedback

Reply:
In f9cfebf I did add a clear interrupt if the interrupt has happened *and* the stack is empty. 

I wondered about moving the `leave` there and that would mean that we also clear the Promises jobs in those cases ... which seems like a thing that needs to happen :thinking:.

But I kind of feel like this needs more thinking about it and definitely even more tests. I can't really see a point if you interrupted to have your promise jobs waiting on you next time you run, and I can't see a way to clear it from outside. So I guess we either need to add one or do it on interrupt either way :thinking:

Reply:
I did think more about it and decided to go ahead and clear the jobqueue as well and update the tests to also use some Promises.

I can't really think of a reason why if you've *interrupted* the Runtime you will want the Promise jobs to be run the next time you use Run* on the same runtime.

Reply:
Looks ok to me. Just thought it would make sense to move the code into a separate method (like `leaveAbrupt()`) and the fact that the job queue is cleared should be mentioned in the documentation.

Reply:
Good idea. I also squashed and reworded the commit to better reflect what the change does now